convert.py

```diff
@@ -15,65 +15,114 @@
+         fetch_targets] = fluid.io.load_inference_model(args.fluid_model, exe)
 
         # Using blocks in programs, create nodes using:
         onnx_nodes = []
-        all_inputs = []
-        for block in inference_program.blocks:
-            all_inputs += [
-                paddle_variable_to_onnx_tensor(v, block) for v in block.vars
-                if v not in ['feed', 'fetch']
-            ]
 
+        # Load parameters
+        global_block = inference_program.global_block()
+        for var_name in global_block.vars:
+            var = global_block.var(var_name)
+            if var_name not in ['feed', 'fetch'] and var.persistable:
+                param = fluid.executor.fetch_var(var_name, inference_scope)
+                param_node = helper.make_node(
+                    'Constant',
+                    inputs=[],
+                    outputs=[var_name],
+                    value=helper.make_tensor(
+                        name=var_name,
+                        dims=var.shape,
+                        data_type=PADDLE_TO_ONNX_DTYPE[var.dtype],
+                        vals=param.flatten().tolist()))
+                onnx_nodes.append(param_node)
+
+        # Create inputs
+        inputs = [
+            paddle_variable_to_onnx_tensor(v, global_block)
+            for v in feed_target_names
+        ]
+
+        # Create outputs
+        fetch_target_names = [
+            fetch_target.name for fetch_target in fetch_targets
+        ]
+        outputs = [
+            paddle_variable_to_onnx_tensor(v, global_block)
+            for v in fetch_target_names
+        ]
+
+        # Create nodes
+        for block in inference_program.blocks:
             for op in block.ops:
-                if op.type in ops.PADDLE_TO_ONNX:
-                    # TODO(varunarora): Attributes.
-                    # TODO(varunarora): Use the modifier function to make the
-                    # transformation.
-                    node_proto = helper.make_node(
-                        ops.PADDLE_TO_ONNX[op.type][0], op.input_arg_names,
-                        op.output_arg_names)
+                if op.type in ops.node_maker:
+                    # TODO(kuke): deal with the corner case that vars in 
+                    #     different blocks have the same name
+                    node_proto = ops.node_maker[op.type](
+                        inputs=op.input_arg_names,
+                        attrs=op.attr_names,
+                        outputs=op.output_arg_names)
 
                     onnx_nodes.append(node_proto)
                 else:
-                    # Not valid to skip any op, so after all edge cases have
-                    # been accounted for, this exception raising to be
-                    # re-enabled.
-                    # raise NameError(op.type)
-                    pass
+                    if op.type not in ['feed', 'fetch']:
+                        raise NotImplementedError("OP[%s] is not supported in "
```

Comment:
Much smarter error than `NameError` :)

Reply:
Yeah, maybe a better one. You remind me to raise an error here ;-).